packages/wallets/src/services/wallet-factory.ts

```diff
@@ -173,12 +189,75 @@ export class WalletFactory {
         }
     }
 
+    /**
```

Comment:
Here, we validate against an existing wallet config:
1) Making sure the signer type is the same 
2) For keypair signers, making sure the signer address is the same